app/lib/theme/CustomFields.tsx

```diff
@@ -0,0 +1,212 @@
+import React from "react";
+import { FieldProps } from "@rjsf/core";
+import NumberFormat from "react-number-format";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faLongArrowAltRight } from "@fortawesome/free-solid-svg-icons";
+
+const showStringDiff = (id: string, oldData: string, newData: string) => (
+  <>
+    <span id={id && `${id}-diffOld`} className="diffOld">
+      {oldData}
+    </span>
+    &nbsp;
+    <FontAwesomeIcon size="lg" color="black" icon={faLongArrowAltRight} />
+    &nbsp;
+    <span id={id && `${id}-diffNew`} className="diffNew">
+      {newData}
+    </span>
+  </>
+);
+
+const showStringAdded = (id: string, newData: string) => (
+  <>
+    <span id={id && `${id}-diffNew`} className="diffNew">
+      {newData}
+    </span>
+    &nbsp;
+    <FontAwesomeIcon size="lg" color="black" icon={faLongArrowAltRight} />
+    &nbsp;
+    <span>
+      <strong>
+        <em>ADDED</em>
+      </strong>
+    </span>
+  </>
+);
+
+const showStringRemoved = (id: string, oldData: string) => (
+  <>
+    <span id={id && `${id}-diffOld`} className="diffOld">
+      {oldData}
+    </span>
+    &nbsp;
+    <FontAwesomeIcon size="lg" color="black" icon={faLongArrowAltRight} />
+    &nbsp;
+    <span>
+      <strong>
+        <em>REMOVED</em>
+      </strong>
+    </span>
+  </>
+);
+
+const showNumberDiff = (
+  id: string,
+  oldData: number,
+  newData: number,
+  isMoney: boolean
+) => (
+  <>
+    <span id={id && `${id}-diffOld`} className="diffOld">
+      <NumberFormat
+        thousandSeparator
+        prefix={isMoney ? "$" : ""}
+        id={id}
+        displayType="text"
+        value={oldData}
+      />
+    </span>
+    &nbsp;
+    <FontAwesomeIcon size="lg" color="black" icon={faLongArrowAltRight} />
+    &nbsp;
+    <span id={id && `${id}-diffNew`} className="diffNew">
+      <NumberFormat
+        thousandSeparator
+        prefix={isMoney ? "$" : ""}
+        id={id}
+        displayType="text"
+        value={newData}
+      />
+    </span>
+  </>
+);
+
+const showNumberAdded = (id: string, newData: number, isMoney: boolean) => (
+  <>
+    <span id={id && `${id}-diffNew`} className="diffNew">
+      <NumberFormat
+        thousandSeparator
+        prefix={isMoney ? "$" : ""}
+        id={id}
+        displayType="text"
+        value={newData}
+      />
+    </span>
+    &nbsp;
+    <FontAwesomeIcon size="lg" color="black" icon={faLongArrowAltRight} />
+    &nbsp;
+    <span>
+      <strong>
+        <em>ADDED</em>
+      </strong>
+    </span>
+  </>
+);
+
+const showNumberRemoved = (id: string, oldData: number, isMoney: boolean) => (
+  <>
+    <span id={id && `${id}-diffOld`} className="diffOld">
+      <NumberFormat
+        thousandSeparator
+        prefix={isMoney ? "$" : ""}
+        id={id}
+        displayType="text"
+        value={oldData}
+      />
+    </span>
+    &nbsp;
+    <span>
+      <strong>
+        <em>REMOVED</em>
+      </strong>
+    </span>
+  </>
+);
+
```

Comment:
We could refactor the different display `<span> {data} / <NUMBER> / ADDED </span>` to avoid repeating them, but I kind of like how readable it is as is